qintelqwatch_connector.py

```diff
@@ -0,0 +1,412 @@
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied. See the License for the specific language governing permissions
+# and limitations under the License.
+#
+
+import json
+import os
+from copy import deepcopy
+from datetime import datetime, timedelta
+
+# Phantom App imports
+import phantom.app as phantom
+from phantom.action_result import ActionResult
+from phantom.base_connector import BaseConnector
+
+from qintelqwatch_consts import *
+from qintelqwatch_helper import search_qwatch
+
+
+class QWatchConnector(BaseConnector):
+
+    def __init__(self):
+
+        # Call the BaseConnectors init first
+        super(QWatchConnector, self).__init__()
+
+        self._state = None
+
+        self.remote = None
+        self.client_id = None
+        self.client_secret = None
+
+    def _validate_integer(self, parameter, key):
+        """
+        Validate an integer.
+
+        :param action_result: Action result or BaseConnector object
+        :param parameter: input parameter
+        :param key: input parameter message key
+        :return: status phantom.APP_ERROR/phantom.APP_SUCCESS, integer value of the parameter or None in case of failure
+        """
+        if parameter is not None:
+            try:
+                if not float(parameter).is_integer():
+                    return self.set_status(phantom.APP_ERROR, VALID_INT_MSG.format(param=key)), None
+
+                parameter = int(parameter)
+            except:
+                return self.set_status(phantom.APP_ERROR, VALID_INT_MSG.format(param=key)), None
+
+            if parameter < 0:
+                return self.set_status(phantom.APP_ERROR, NON_NEG_INT_MSG.format(param=key)), None
+
+        return phantom.APP_SUCCESS, parameter
+
+    def _handle_test_connectivity(self):
+
+        action_result = self.add_action_result(ActionResult())
+        try:
+            res = search_qwatch(None, None, 'ping', **self.client_args)
+            self.debug_print(f'qwatch test connectivity return: {res}')
+        except Exception as e:
+            return action_result.set_status(phantom.APP_ERROR,
+                            f'Test Connectivity Failed {str(e)}')
+
+        self.save_progress("Test Connectivity Passed")
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _qwatch_query(self, search_term, search_type, query_type, search_args):
+
+        kwargs = deepcopy(self.client_args)
+        kwargs.update(search_args)
+
+        try:
+            return search_qwatch(
+                search_term,
+                search_type,
+                query_type,
+                **kwargs
+            )
+        except Exception as e:
+            self.debug_print(f'qwatch lookup failed: {str(e)}')
+            raise Exception(str(e))
+
+    def _make_qwatch_times(self, params):
+
```

Comment:
Could you add docstring for this one as well? It's not clear what qwatch times are.

Reply:
Done